homeassistant/components/blink/switch.py

```diff
@@ -0,0 +1,106 @@
+"""Support for Blink Motion detection switches."""
+from __future__ import annotations
+
+import asyncio
+from typing import Any
+
+from homeassistant.components.switch import (
+    SwitchDeviceClass,
+    SwitchEntity,
+    SwitchEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DEFAULT_BRAND, DOMAIN, TYPE_CAMERA_ARMED
+from .coordinator import BlinkUpdateCoordinator
+
+SWITCH_TYPES: tuple[SwitchEntityDescription, ...] = (
+    SwitchEntityDescription(
+        key=TYPE_CAMERA_ARMED,
+        icon="mdi:motion-sensor",
+        translation_key="camera_motion",
+        device_class=SwitchDeviceClass.SWITCH,
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Blink switches."""
+    coordinator: BlinkUpdateCoordinator = hass.data[DOMAIN][config.entry_id]
+
+    entities = [
+        BlinkSwitch(coordinator, camera, description)
+        for camera in coordinator.api.cameras
+        for description in SWITCH_TYPES
+    ]
+    async_add_entities(entities)
+
+
+class BlinkSwitch(CoordinatorEntity[BlinkUpdateCoordinator], SwitchEntity):
+    """Representation of a Blink motion detection switch."""
+
+    _attr_has_entity_name = True
+    _attr_name = None
```

Comment:
Having both a translation key and `_attr_name = None` doesn't make a lot of sense